pkg/tests/tasks/traffic/egress/egress_gateways_tls_file_mount_test.go

```diff
@@ -15,173 +15,198 @@
-
-	defer cleanupTLSOriginationFileMount()
-	defer util.RecoverPanic(t)
-
-	log.Log.Info("TestEgressGatewaysTLSOrigination File Mount")
-	sleep := examples.Sleep{Namespace: "bookinfo"}
-	sleep.Install()
-	sleepPod, err := util.GetPodName("bookinfo", "app=sleep")
-	util.Inspect(err, "Failed to get sleep pod name", "", t)
-
-	t.Run("TrafficManagement_egress_gateway_perform_TLS_origination", func(t *testing.T) {
-		defer util.RecoverPanic(t)
-
-		log.Log.Info("Perform TLS origination with an egress gateway")
-		util.KubeApplyContents("bookinfo", ExServiceEntry)
-		time.Sleep(time.Duration(10) * time.Second)
-
-		command := `curl -sSL -o /dev/null -D - http://istio.io`
-		msg, err := util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "301 Moved Permanently") {
-			log.Log.Info("Success. Get http://istio.io response")
-		} else {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		}
-
-		log.Log.Info("Create a Gateway to external istio.io")
-		util.KubeApplyContents("bookinfo", util.RunTemplate(ExGatewayTLSFileTemplate, smcp))
-		time.Sleep(time.Duration(20) * time.Second)
-
-		command = `curl -sSL -o /dev/null -D - http://istio.io`
-		msg, err = util.PodExec("bookinfo", sleepPod, "sleep", command, false)
-		util.Inspect(err, "Failed to get response", "", t)
-		if strings.Contains(msg, "301 Moved Permanently") || !strings.Contains(msg, "200") {
-			log.Log.Infof("Error response: %s", msg)
-			t.Errorf("Error response: %s", msg)
-		} else {
-			log.Log.Infof("Success. Get http://istio.io response")
-		}
-
-		log.Log.Info("Cleanup the TLS origination example")
-		util.KubeDeleteContents("bookinfo", util.RunTemplate(ExGatewayTLSFileTemplate, smcp))
-		util.KubeDeleteContents("bookinfo", ExServiceEntry)
-		time.Sleep(time.Duration(20) * time.Second)
-	})
+	NewTest(t).Id("T14").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		t.Log("Test TLS Origination with File Mount")
+		ns := "bookinfo"
+
+		app.InstallAndWaitReady(t, app.Sleep(ns))
+
+		t.NewSubTest("Traffic Management egress gateway TLS origination").Run(func(t TestHelper) {
+			t.Log("Perform TLS origination with an egress gateway")
+			t.Cleanup(func() {
+				oc.DeleteFromTemplate(t, ns, ExGatewayTLSFileTemplate, smcp)
+				oc.DeleteFromString(t, ns, ExServiceEntry)
+			})
+
+			t.LogStep("Create ServiceEntry for port 80 and 443 and verify that the egress gateway is working: expect 301 Moved Permanently from istio.io")
+			oc.ApplyString(t, ns, ExServiceEntry)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=sleep", ns),
+					"sleep", `curl -sSL -o /dev/null -D - http://istio.io`,
+					assert.OutputContains(
+						"301",
+						"Expected 301 Moved Permanently",
+						"ERROR: Not expected response, expected 301 Moved Permanently"))
```

Comment:
Instead of checkiing for 301, can you change this so it checks for 200, just like we've done in some other tests. For example here: https://github.com/maistra/maistra-test-tool/pull/444/files#diff-19d3cba61e66345fad4a9ce62937199acb7c03427483a21cac9c8a9a337faf7bR49-R52

You will have to modify the curl command so that it returns httpCode instead of using `-D -`.

Reply:
This could also be made into a `assertRequestSuccess` and re-used in line 63 of this test. See: https://github.com/maistra/maistra-test-tool/pull/444/files#diff-19d3cba61e66345fad4a9ce62937199acb7c03427483a21cac9c8a9a337faf7bR46-R53

Reply:
BTW: in some tests, we honor the `HTTP_PROXY` environment variable, but not in all. Do you know why we added that? Why just in some tests and not everywhere? 

I don't think it makes sense to look at the local `HTTP_PROXY` variable and then use its value within the sleep pod. That pod might not need to use a proxy.

Reply:
HTTP_PROXY This variable was added to work with IPv6 cluster because Filip set a PROXY to make the IPv4 to IPv6 translation

Reply:
Oh, so that means the proxy must be used in all tests. Maybe that's why the tests are failing?

Reply:
Could be, but for some reason I test in a point explicit setting the variable and does not work either

Reply:
The PROXY was only used for http. What about https?

Reply:
Reading again this, we need to check that is a 301 code because is an automatic redirection from http to https in istio.io. Makes sense to force to a 200 here?

Reply:
this test is weird if it uses the difference between the status codes (301 vs 200) to determine whether the traffic is being routed through the correct component... we should fix this some time.